### ISSUE TITLE
Nested relations - manual search one column - id issues

### ISSUE THREAD

**Issue by apsg** [yajra/laravel-datatables#1019]
I'm trying to make a custom input field that allows user to search the value in one of the table's columns. Unfortunately this column is a nested relation.  

```
 columns: [
            { data: 'id', name: 'id', searchable: true },
            { data: 'date', name: 'date', searchable : true },
            { data: 'atlas', name: 'atlas', searchable:true },
            { data: 'client.name', name: 'client.name' , searchable: true },
            { data: 'qr.deviceid', name: 'qr.deviceid' , searchable : true}
]
```

On my input field I attach the onchange event like that:

```
$('.search').on('change', function(e){
            e.preventDefault();
            table.column( $(this).data('columnid') )
                 .search( $(this).val() )
                 .draw();
        });
```
where `columnid` represents the ID of searched column. 

If columnid represents "normal" column, like date or atlas, the search works perfectly fine. But if I try to search the client.name column, I get the following error:

```
QueryException in Connection.php line 647:
SQLSTATE[42S21]: Column already exists: 1060 Column duplicate name 'id' (SQL: select count(*) as aggregate from (select * from `forms` left join `clients` on `forms`.`client_id` = `clients`.`id` where `send_to_financial` is not null and `is_paid` is not null and LOWER(`clients`.`name`) LIKE %%sd a%% group by IFNULL(forms.guid, forms.id)) count_row_table)
```

How can I make this custom one-column-search work? Is it a bug or am I doing something wrong?

**Comment by minedun6** [comment]
A good way of going around I think is to alias the columns you return from executing your SQL query.
And one last tip, don't use eloquent when working with this package as it struggles with resolving relationships, if you do any sorting on your table, it'll tell u sometimes that it can't find a column.
Instead favor the query builder.

**Comment by apsg** [comment]
Since I'm using scopes and other stuff, I really cannot switch to pure sql right now. The main question is why all works in normal sorting/searching through all columns, but when I specify the column it crashes.

**Comment by yajra** [comment]
@apsg try adding the table name on your scripts:

```
 { data: 'id', name: 'forms.id', searchable: true },
...
```

**Comment by apsg** [comment]
@yajra - this change breaks the whole column and throws error:

> DataTables warning: table id=forms-table - Requested unknown parameter 'form.id' for row 0, column 0. For more information about this error, please see http://datatables.net/tn/4

**Comment by apsg** [comment]
@yajra - maybe there is other way of enabling the search of only one column?

**Comment by yajra** [comment]
I think this is already resolved on #1046? Thanks!
